mailet/api/src/main/java/org/apache/mailet/FutureReleaseParameters.java

```diff
@@ -0,0 +1,130 @@
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.mailet;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+public class FutureReleaseParameters {
+    public static final String HOLDFOR_PARAMETER = "HOLDFOR";
+    public static final String HOLDUNITL_PARAMETER = "HOLDUNITL";
+    public static final long holdforValue = 86400;
+
+    public static class Holdfor {
+        public static Optional<Holdfor> fromSMTPArgLine(Map<String, Long> mailFromArgLine) {
+            return Optional.ofNullable(mailFromArgLine.get(HOLDFOR_PARAMETER))
+                .map(Holdfor::of);
+        }
+
+        public static Holdfor fromAttributeValue(AttributeValue<Long> attributeValue) {
+            return of(attributeValue.value());
+        }
+
+        private final Long value;
+
+        public static Holdfor of(Long value) {
+            Preconditions.checkNotNull(value);
+            return new Holdfor(value);
+        }
+
+        private Holdfor(Long value) {
+            this.value = value;
+        }
+
+        public Long asString() {
+            return value;
+        }
+
+        public AttributeValue<Long> toAttributeValue() {
+            return AttributeValue.of(value);
+        }
+
+        @Override
+        public final boolean equals(Object o) {
+            if (o instanceof Holdfor) {
+                Holdfor holdfor = (Holdfor) o;
+                return Objects.equals(this.value, holdfor.value);
+            }
+            return false;
+        }
+
+        @Override
+        public final int hashCode() {
+            return Objects.hash(value);
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                .add("value", value)
+                .toString();
+        }
+    }
+
+    public static class Holduntil {
```

Comment:
```suggestion
    public static class HoldUntil {
```

tip: use IntelliJ refactoring feature to easily refactor the class name.